week2/409_longest_palindrome.py

```diff
@@ -0,0 +1,29 @@
+# https://leetcode.com/problems/longest-palindrome/description/
+from collections import defaultdict
+
+
+def longest_palindrome(self, s: str) -> int:
+    if len(s) == 1:
+        return 1
+
+        # create frequency dictionary
+    freq_dist = defaultdict(int)
+    for letter in s:
+        if letter in freq_dist:
+            freq_dist[letter] += 1
+        else:
+            freq_dist[letter] = 1
+
+    if len(freq_dist.values()) == len(s):
```

Comment:
querying values will cause unnecessary iteration over `freq_dist`. you can just do `len(freq_dist)` 
I guess this condition happens rarely, and doesn't help much in computation reduction, leaving this out would be more simpler.